src/main/java/space/space_spring/domain/pay/model/firstCollection/PayRequestTargets.java

```diff
@@ -0,0 +1,40 @@
+package space.space_spring.domain.pay.model.firstCollection;
+
+import jakarta.persistence.Embeddable;
+import jakarta.persistence.Embedded;
+import lombok.Builder;
+import lombok.Getter;
+import space.space_spring.domain.pay.model.dto.PayTargetInfoDto;
+import space.space_spring.domain.pay.model.entity.PayRequestTarget;
+
+import java.util.List;
+
+@Getter
+public class PayRequestTargets {
+
+    private final List<PayRequestTarget> payRequestTargets;
+
+    @Builder
+    private PayRequestTargets(List<PayRequestTarget> payRequestTargets) {
+        this.payRequestTargets = payRequestTargets;
+    }
+
+    public static PayRequestTargets create(List<PayRequestTarget> payRequestTargets) {
+        return PayRequestTargets.builder()
+                .payRequestTargets(payRequestTargets)
+                .build();
+    }
+
+    public PayTargetInfos getPayTargetInfos() {
+        PayTargetInfos payTargetInfos = new PayTargetInfos();
+
+        for (PayRequestTarget payRequestTarget : payRequestTargets) {
+            PayTargetInfoDto payTargetInfo = payRequestTarget.createPayTargetInfo();
+            payTargetInfos.add(payTargetInfo);
+        }
+
+        return payTargetInfos;
```

Comment:
마찬가지로 불변성을 위해 `PayRequestTarget`으로 `List`를 생성한 뒤 `PayRequestTargets` 생성자를 통해 일급컬렉션을 초기화하는 방식이 좋을 것 같습니다!

Reply:
일급컬렉션은 비즈니스 로직의 응집성을 높이기 위해 도입한 방법으로 알고 있어요. 따라서 일급컬렉션의 멤버 변수는 생성 후 외부에서의 추가/삭제 등의 수정 없이 불변성이 유지되며, 일급컬렉션 내부에서의 로직 수행만 허용한다고 합니다. 
현재 코드상으로는 payTargetInfos.add를 통해 직접 객체를 추가하고 있다는 점을 말씀드린 거예요!
```suggestion
      return PayTargetInfos.from(payRequestTargets);
```
```
public class PayTargetInfos {
    private final List<PayTargetInfoDto> infos;

    public PayTargetInfos(List<PayTargetInfoDto> infos) {
        this.infos = new ArrayList<>(infos);
    }

    public static PayTargetInfos from(List<PayRequestTarget> targets) {
        return new PayTargetInfos(
            targets.stream()
                   .map(PayRequestTarget::createPayTargetInfo)
                   .toList()
        );
}
```
이건 위 리뷰에서 말씀드린 것과는 다른 방법이긴 한데, 정적 팩토리 메소드를 사용해서 생성과 동시에 값을 설정할 수도 있을 것 같아요. 이런 방식으로 일급컬렉션의 장점을 더 활용할 수 있지 않을까요?

Reply:
앗 직접 list.add()를 하는 것이 아니라 제가 작성한 방식처럼
payTargetInfos.add 를 통해서 PayTargetInfos 일급 컬렉션에게 메시지를 전달하고, 일급 컬렉션은 add 메서드를 통해서 자신의 필드(list) 에 전달받은 인자를 추가하는 방식도 외부에서 값을 추가한다고 보여지는 걸까요???

Reply:
음 말씀하신 방법처럼 정적 팩토리 메서드를 활용해 인자로 받은 컬렉션으로 일급 컬렉션 객체를 생성한 후, 해당 컬렉션을 불변한 상태 (final) 로 만들고 나서 비즈니스 로직을 수행하는 방법도 좋지만,
도메인 로직에 따라 나중에 추가되는 요소들이 있을 수 있고, 또한 내부 컬렉션에 어떤 요소를 추가하고 제거하는 것도 일급컬렉션 차원에서 필요한 비즈니스로직에 포함되는 거 아닌가? 라는 생각도 있습니다.

그리고 이렇게 일급 컬렉션에 add, remove 같은 메서드를 열어 둔다고 해서 외부에서 일급 컬렉션에게 add, remove 라는 메시지를 전달하지 않고 직접 해당 컬렉션을 수정할 수 있는 방법이 없으므로 이것 또한 컬렉션을 불변한 상태로 유지한다는 일급 컬렉션의 정책에 맞지않나? 라는 생각도 드네여

뭔가 정답이 없는것같다? 라는 게 제 생각이긴 합니다

추가로 일급 컬렉션에 add, remove 메서드를 구현한 예제 링크 하나 달아놓겠습니닷
https://blogshine.tistory.com/475

그래도 덕분에 이번 기회에 일급 컬렉션 제대로 공부하는 거 같네여 허허